django/views/generic/detail.py

```diff
@@ -19,6 +19,20 @@ class SingleObjectMixin(ContextMixin):
     pk_url_kwarg = 'pk'
     query_pk_and_slug = False
 
+    def get_model(self):
+        """
+        Return the model of the object the view is displaying.
+
+        By default, this returns `self.model`, but subclasses can override
+        this and return any model
+        """
+        if self.model is None:
+            raise ImproperlyConfigured(
+                "SingleObjectMixin requireds either a definition of "
+                "'model' or an implementation of 'get model()'")
```

Comment:
A message similar to the one raised in `get_object()` would be more appropriate:

```python
raise ImproperlyConfigured(
    "%(cls)s is missing a Model. Define %(cls)s.model or "
    "override %(cls)s.get_model()." % {
        'cls': self.__class__.__name__
    }
)
```